lib/cli/src/Cardano/CLI.hs

```diff
@@ -298,6 +319,147 @@ runCli = join . customExecParser preferences
   where
     preferences = prefs showHelpOnEmpty
 
+{-------------------------------------------------------------------------------
+                            Commands - 'HD Derivation'
+-------------------------------------------------------------------------------}
+
+cmdKey :: Mod CommandFields (IO ())
+cmdKey = command "key" $ info (helper <*> cmds) $ mempty
+    <> progDesc "Derive keys from mnemonics."
+  where
+    cmds = subparser $ mempty
+        <> cmdRootKey
+
+-- | Parse a key type
+--
+-- Note that we in the future might replace the type @ByronWalletStyle@ with
+-- another type, to include Shelley keys.
+parseKeyType :: Parser (IO ByronWalletStyle)
+parseKeyType = toIO . fromText <$> strOption
+    ( long "type"
+    <> metavar "KEYTYPE"
+    <> helpDoc (Just (vsep typeOptions))
+    )
+  where
+    toIO :: Either TextDecodingError ByronWalletStyle -> IO ByronWalletStyle
+    toIO (Right x) =
+        return x
+    toIO (Left (TextDecodingError e)) =
+        hPutStrLn stderr e >> exitFailure
+
+    typeOptions = string <$>
+        [ "Any of the following:"
+        , "  random (Daedalus, " ++ allowedWords Random ++ ")"
```

Comment:
We should not mention Daedalus here because this wallet scheme is not specific to Daedalus. Exchanges are using it as well. Stick with `random` :+1:

Reply:
A fleeting concern I had was that one _might_ think that that `random` meant "choose one of the other three schemes at random" 🙈

Reply:
10c1378dbc9f858455fde1d5bb52c5e0fa69c70e